src/core/unit.h

```diff
@@ -117,7 +117,8 @@ typedef struct Unit {
         FreezerState freezer_state;
         sd_bus_message *pending_freezer_message;
 
-        char *id; /* One name is special because we use it for identification. Points to an entry in the names set */
+        char *id; /* One name is special because we use it for identification. If names set is defined, this
+                   * pointer points to an entry there. Otherwise, this is the only name. */
```

Comment:
maybe extend this, to clarify the memory management here, since it's non obvious. i.e. that if there's a set then id points into the set and doesn't need to be freed separately, but if there's no set, then id needs to be freed.

Reply:
This part is reworked now.